src/pages/ProductsPage/ProductsPage.tsx

```diff
@@ -0,0 +1,37 @@
+import { Box } from '@mui/material';
+import Button from '@mui/material/Button';
+import { GridToolbar} from '@mui/x-data-grid/components/toolbar';
+import { DataGrid } from '@mui/x-data-grid/DataGrid';
+import { productTableColumns } from '../../constants/productTableColumns';
+import { ProductModel } from '../../models/ProductModel';
+
+interface Props {
+  getProductsClick: () => void;
+  products: { [productId: number]: ProductModel };
+  productIds: number[];
+}
+
+export default function ProductsPage (props: Props) {
+  return (
+    <>
+        <Box textAlign="center" marginBottom = '8px'>
+          <Button variant="contained" onClick={props.getProductsClick} size = 'large'>
+            Get Products
+          </Button>
+        </Box>
+        <Box sx={{ height: 400, width: '100%' }}>
+          <DataGrid
+            columns={productTableColumns}
+            rows={Object.values(props.products)}
```

Comment:
remove Object.values(props.products). Pass just a props.products (or rows, any valid mapped model for rows). Map it the container.